src/dataExplorer/context/fluxQueryBuilder.tsx

```diff
@@ -95,6 +124,39 @@ export const FluxQueryBuilderProvider: FC = ({children}) => {
     getTagKeys(selection.bucket, measurement)
   }
 
+  const handleSelectField = (field: string): void => {
+    // Inject field
+    injectViaLsp(ExecuteCommand.InjectField, {
+      bucket:
```

Comment:
why do we need the bucket here? isn't that just used on measurement injection for the current iteration?

Reply:
It's required in LSP https://github.com/influxdata/flux-lsp/blob/master/src/server/mod.rs#L1333-L1336

Here is my understanding:

No matter the user is injecting measurement, field, or tag value in UI, LSP treats them as injecting `filter()` function. In order to make sure the injection line works, LSP asks for the bucket that filter function is applying to. 

Or there are other intentions? Correct me if I understand it wrong @rockstar

Reply:
Append only, to the last flux pipeline in the query.
Confirms that the last flux transformation pipeline has the same bucket as we are seeking.

This is the first iteration of the injection in the LSP.
Plans are already in the works for the next one.